src/drivers/mediatek/mt8195/ipc.c

```diff
@@ -127,6 +127,10 @@ int platform_ipc_init(struct ipc *ipc)
 	struct ipc_data *iipc;
 
 	iipc = rzalloc(SOF_MEM_FLAG_KERNEL, sizeof(*iipc));
+	if (!iipc) {
+		tr_err(&ipc_tr, "Unable to allocate memory for IPC data");
+		return -ENOMEM;
```

Comment:
somehow I regularly end up reviewing from bottom to top. Same comment as below - panic?